pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -553,6 +557,10 @@ func (r *repository) rememberLayersOfImage(image *imageapi.Image, cacheName stri
 		for _, layer := range image.DockerImageLayers {
 			r.cachedLayers.RememberDigest(digest.Digest(layer.Name), r.blobrepositorycachettl, cacheName)
 		}
+		// remember reference to manifest config as well for schema 2
+		if image.DockerImageManifestMediaType == schema2.MediaTypeManifest && len(image.DockerImageMetadata.ID) > 0 {
```

Comment:
> should we only do this if len(image.DockerImageConfig) > 0?

To make it comply with the other conditions, I'll use this instead. `image.DockerImageConfig` is filled only for manifest schema 2.